tests/testsuite/update.rs

```diff
@@ -2064,3 +2112,507 @@ fn update_breaking_specific_packages() {
+        .with_stderr_data(str![[r#"
+[UPDATING] `[..]` index
+
+"#]])
+        .run();
+
+    let crate_manifest_after = p.read_file("Cargo.toml");
+    assert_e2e().eq(&crate_manifest_after, crate_manifest);
+
+    let lock_file_after = p.read_file("Cargo.lock");
+    assert_e2e().eq(&lock_file_after, lock_file);
+
+    p.cargo(
+        "update compatible renamed-from non-semver transitive-compatible transitive-incompatible",
+    )
+    .with_stderr_data(str![[r#"
+[UPDATING] `[..]` index
+[LOCKING] 5 packages to latest compatible versions
+[UPDATING] compatible v1.0.0 -> v1.0.1
+[UPDATING] non-semver v1.0.0 -> v1.0.1 (latest: v2.0.0)
+[UPDATING] renamed-from v1.0.0 -> v1.0.1 (latest: v2.0.0)
+[UPDATING] transitive-compatible v1.0.0 -> v1.0.1
+[UPDATING] transitive-incompatible v1.0.0 -> v1.0.1 (latest: v2.0.0)
+
+"#]])
+    .run();
+}
+
+#[cargo_test]
+fn update_breaking_without_lock_file() {
+    Package::new("compatible", "1.0.0").publish();
+    Package::new("incompatible", "1.0.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name  =  "foo"
+            version  =  "0.0.1"
+            edition  =  "2015"
+            authors  =  []
+
+            [dependencies]
+            compatible  =  "1.0"  # Comment
+            incompatible  =  "1.0"  # Comment
+        "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    Package::new("compatible", "1.0.1").publish();
+    Package::new("incompatible", "1.0.1").publish();
+
+    Package::new("incompatible", "2.0.0").publish();
+
+    p.cargo("update -Zunstable-options --breaking")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_stderr_data(str![[r#"
+[UPDATING] `[..]` index
+[UPGRADING] incompatible ^1.0 -> ^2.0
+[LOCKING] 3 packages to latest compatible versions
+
+"#]])
+        .run();
+}
+
+#[cargo_test]
+fn update_breaking_spec_version() {
+    Package::new("compatible", "1.0.0").publish();
+    Package::new("incompatible", "1.0.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name  =  "foo"
+            version  =  "0.0.1"
+            edition  =  "2015"
+            authors  =  []
+
+            [dependencies]
+            compatible  =  "1.0"  # Comment
+            incompatible  =  "1.0"  # Comment
+        "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("generate-lockfile").run();
+
+    Package::new("compatible", "1.0.1").publish();
+    Package::new("incompatible", "1.0.1").publish();
+
+    Package::new("incompatible", "2.0.0").publish();
+
+    // Invalid spec
+    p.cargo("update -Zunstable-options --breaking incompatible@foo")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_status(101)
+        .with_stderr_data(str![[r#"
+[ERROR] expected a version like "1.32"
```

Comment:
Hadn't notice this in your prior PR.  This error message is bad.  We should at least be calling out that we are parsing a package id spec

Existing behavior:
```console
$ cargo update clap@foo
error: invalid package ID specification: `clap@foo`

Caused by:
  expected a version like "1.32"
```
(again, task is list is fine)

Reply:
Added.